apps/homework/views.py

```diff
@@ -54,53 +54,159 @@ def get_context_data(self, **kwargs):
+                return True
+        except ObjectDoesNotExist:
+            return False
+        return False
+
+class SubmissionListView(LoginRequiredMixin, UserPassesTestMixin, TemplateView):
+    template_name = 'homework/submission_list.html'
+
+    def get_context_data(self, **kwargs):
+        context = super(SubmissionListView, self).get_context_data(**kwargs)
+        definition_pk = self.kwargs.get('definition_pk')
+        submissions = None
+        try:
+            definition = Definition.objects.get(pk=definition_pk)
+            context['definition'] = definition
+            context['klass'] = definition.klass
+        except ObjectDoesNotExist:
+            raise Http404('Definition object not found')
+
+        # if user is instructor of the class
+        if self.request.user.instructor and self.request.user.instructor == definition.klass.instructor:
+            try:
+                submissions = Submission.objects.filter(definition=definition)
+                context['instructor_view'] = True
+            except ObjectDoesNotExist:
+                raise Http404('Submission object not found')
+
+        # if user is member of the class
+        else:
+            try:
+                klass_membership = self.request.user.klass_memberships.get(klass__pk=definition.klass.pk)
+            except ObjectDoesNotExist:
+                raise Http404('Student Membership object not found')
+            try:
+                if definition.team_based:
+                    team = klass_membership.team
+                    submissions = Submission.objects.filter(definition=definition, team=team)
+                    context['team'] = team
+                else:
+                    submissions = Submission.objects.filter(definition=definition, creator=klass_membership)
+            except ObjectDoesNotExist:
+                raise Http404('Submission object not found')
+        context['submissions'] = submissions
+        context['user'] = self.request.user
+        return context
+
+    def test_func(self):
+        definition_pk = self.kwargs.get('definition_pk')
+        try:
+            definition = Definition.objects.get(pk=definition_pk)
+        except ObjectDoesNotExist:
+            raise Http404('Definition object not found')
+        if self.request.user.is_superuser:
+            return True
+        if self.request.user.instructor:
+            if self.request.user.instructor == definition.klass.instructor:
+                return True
+        try:
+            if self.request.user.klass_memberships.get(klass__pk=definition.klass.pk):
+                return True
+        except ObjectDoesNotExist:
+            return False
+        return False
+
 class SubmissionFormView(LoginRequiredMixin, TemplateView):
     template_name = 'homework/forms/submit.html'
 
-    def get_context_data(self, **kwargs):
+    def get_context_data(self, use_github=False, **kwargs):
```

Comment:
Shouldn't `use_github` already be in `kwargs` or `self.kwargs`?

Reply:
You are correct, sir. Thank you!